multiplier_analysis/nbconverted/0_process_multiplier_model.py

```diff
@@ -0,0 +1,135 @@
+
+# coding: utf-8
+
+# # Process MULTIPLIER model data
+# 
+# Raw multiplier model data can be found [here](https://github.com/greenelab/multi-plier).
+# 
+# This multiplier model (Robject) was formatted in python in [phenoplier repo](https://github.com/greenelab/phenoplier/blob/master/nbs/01_preprocessing/005-multiplier_recount2_models.ipynb). The python pickled data files were output to a [shared google drive](https://drive.google.com/drive/folders/12wvqGzFpFBUOX_CsFkFvvcbKfhl648LE). Since these pickle files were generated using a different conda environment, this notebook is loading the data from the pickle files into .tsv files for use in our analysis.
+
+# In[1]:
+
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+import os
+import pandas as pd
+import pickle
+
+import rpy2.robjects as ro
+from rpy2.robjects import pandas2ri
+from rpy2.robjects.conversion import localconverter
+
+
+# In[2]:
+
+
+multiplier_dir = os.path.join("/home/alexandra/Documents/Data/Generic_expression_patterns/", "multiplier")
+
+
+# # Load pickle files and save as .tsv
+
+# In[3]:
+
+
+multiplier_u_filename = os.path.join(multiplier_dir, "multiplier_model_u.pkl")
+multiplier_z_filename = os.path.join(multiplier_dir, "multiplier_model_z.pkl")
+
+
+# In[4]:
+
+
+# Load pickled file
+with open(multiplier_u_filename, "rb") as scaler_fh:
+    multiplier_model_u = pickle.load(scaler_fh)
+
+with open(multiplier_z_filename, "rb") as scaler_fh:
+    multiplier_model_z = pickle.load(scaler_fh)
+
+
+# In[5]:
+
+
+print(multiplier_model_u.shape)
+multiplier_model_u.head()
+
+
+# In[6]:
+
+
+print(multiplier_model_z.shape)
+multiplier_model_z.head()
+
+
+# In[7]:
+
+
+# make sure I'm seeing the same when loaded with R
+assert multiplier_model_z.loc['GAS6', 'LV2'] == 0
+assert multiplier_model_z.loc['GAS6', 'LV3'] == 0.039437739697954444
+assert multiplier_model_z.loc['GAS6', 'LV984'] == 0.3473620915326928
+assert multiplier_model_z.loc['GAS6', 'LV987'] == 0
+
+assert multiplier_model_z.loc['SPARC', 'LV981'] == 0
+assert multiplier_model_z.loc['SPARC', 'LV986'].round(8) == 0.12241734
+
+
+# In[8]:
+
+
+# Save
+multiplier_model_u.to_csv("multiplier_model_u.tsv", sep="\t")
+multiplier_model_z.to_csv("multiplier_model_z.tsv", sep="\t")
+
+
+# # Format multiplier summary data
+
+# In[9]:
+
+
+readRDS = ro.r['readRDS']
+
+
+# In[10]:
+
+
+multiplier_full_model = readRDS(os.path.join(multiplier_dir,
```

Comment:
Since you are using Python 3.7, in the future, I found the pathlib module great to handle paths in a cross-platform way (thanks again @danich1 !). You can replace `os.path.join` with the `Path` class, which provides very nice extra methods.

Reply:
This is a great point! I started playing around with using this paths.py file after seeing other greenelab member's repos - its definitely a lot more organized for defining paths. The reason I didn't end up using it for this repo is that I didn't think it was intuitive to the user to change the python file to import if they wanted to switch configurations. Also, at the time, I didn't think you could define both file paths and variables in the same script which made this direction seem less clean.

Since this repo is getting wrapped up, I will likely not make this refactor. But something I will definitely try to use in my next repo